tools/harness-simulation/harness/Thread_Harness/THCI/OpenThread_Sim.py

```diff
@@ -71,26 +76,21 @@ def __connect(self, device, node_id):
             else:
                 raise Exception('Password error')
 
-        self.__stdin, self.__stdout, _ = self.__handle.exec_command(device + ' ' + str(node_id), get_pty=True)
-        self.__stdout.channel.setblocking(0)
-
-        # Wait some time for initiation
-        time.sleep(0.1)
+        self.__handle.get_transport().set_keepalive(self.KEEPALIVE_INTERVAL)
+        self.__stdin, self.__stdout, _ = self.__handle.exec_command(device + ' ' + str(node_id))
+        self.__stdout.channel.settimeout(self.timeout)
```

Comment:
Could you elaborate more on why set timeout and blocking?

Reply:
It receives the output in blocking mode with a timeout.
Setting to the blocking mode is to avoid loop of sleeping and checking outputs. We can determine whether an output is complete or not by checking if the line starts with `Done`, `Error <error_code>: `, etc.
Setting the timeout is for robustness in case of being blocked by unknown errors.